capstone/src/main/webapp/sheetsScript.js

```diff
@@ -62,29 +64,282 @@ async function gatherSheetsInput() {
     }
   }
 
-  const text = await getTextFromSheet(id);
-  handleInput(text, langElement.value, requestedAttributes, tokenizer);
+  // Show general output
+  let sheetNames = await getSheetNames(id);
+  let totalText = '';
+  let range = '!A1:YY';
+
+  // Check if user has entered a custom range
+  const userRangeElement = document.getElementById('sheets-range-yes-no');
```

Comment:
This code is a little hard to follow because above you do `let sheetNames = await getSheetNames(id);`, but then you don't use that value right away and instead possibly overwrite sheetNames based on the user input.

Would it maybe make sense to combine all of this logic for the sheetNames and range into some separate function?  Also maybe if the user specifies something in the `sheets-range` element, then you don't need to make the sheets API request, which might be a little more efficient?

Reply:
That makes sense. I'll refactor this into a separate function that returns `sheetNames` and `range` and reorganize the logic.